web3/utils/abi.py

```diff
@@ -111,6 +116,52 @@ def filter_by_argument_name(argument_names, contract_abi):
     ]
 
 
+if LooseVersion(eth_abi.__version__) >= LooseVersion("0.5.0"):
+    def process_type(type_str):
+        normalized_type_str = normalize_type_string(type_str)
+        abi_type = parse_type_string(normalized_type_str)
+
+        type_str_repr = repr(type_str)
+        if type_str != normalized_type_str:
+            type_str_repr = '{} (normalized to {})'.format(
+                type_str_repr,
+                repr(normalized_type_str),
+            )
+
+        if isinstance(abi_type, TupleType):
+            raise ValueError(
+                "Cannot process type {}: tuple types not supported".format(
+                    type_str_repr,
+                )
+            )
+
+        abi_type.validate()
+
+        sub = abi_type.sub
+        if isinstance(sub, tuple):
+            sub = 'x'.join(map(str, sub))
+        elif isinstance(sub, int):
+            sub = str(sub)
+        else:
+            sub = ''
+
+        arrlist = abi_type.arrlist
+        if isinstance(arrlist, tuple):
+            arrlist = list(map(list, arrlist))
+        else:
+            arrlist = []
+
+        return abi_type.base, sub, arrlist
+
+    def collapse_type(base, sub, arrlist):
+        return base + str(sub) + ''.join(map(repr, arrlist))
+else:
+    from eth_abi.abi import (
+        process_type,
+        collapse_type,
+    )
+
+
```

Comment:
@carver Something like this?

Reply:
Yup, although I think it should be >= v1 in the `if` test. It highlights the breaking change at the major version. (Also, 0.5.0 is already released, and we could theoretically release an 0.6.0 on the v0/stable branch).